types/feegrant.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import feegranttypes "github.com/cosmos/cosmos-sdk/x/feegrant"
+
+// FeeGrant represents the x/feegrant module
+type FeeGrant struct {
+	*feegranttypes.MsgGrantAllowance
```

Comment:
I think that instead of having a `MsgGrantAllowance` here it would be better to use the correct type that identifies a grant: `feegrant.Grant`. This would make sure we are dealing with the same types used by the Cosmos SDK to store the data